src/main/java/laddergame/domain/game/GameResult.java

```diff
@@ -0,0 +1,33 @@
+package laddergame.domain.game;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import laddergame.domain.LadderResultItem;
+import laddergame.domain.PersonalName;
+
+public class GameResult {
+    final private Map<PersonalName, LadderResultItem> nameToItem;
+
+    public GameResult(Map<PersonalName, LadderResultItem> nameToItem) {
+        this.nameToItem = nameToItem;
+    }
+
+    public LadderResultItem searchBy(String value) {
+        PersonalName keyName = PersonalName.valueOf(value);
+        if (!nameToItem.containsKey(keyName)) {
+            throw new IllegalArgumentException("게임 참여자의 이름이 아닙니다.");
+        }
+        return nameToItem.get(keyName);
+    }
```

Comment:
LadderResultItem -> getLadderResultItem

Reply:
```suggestion
    public LadderResultItem getLadderResultItemBy(String value) {
        PersonalName keyName = PersonalName.valueOf(value);
        if (!nameToItem.containsKey(keyName)) {
            throw new IllegalArgumentException("게임 참여자의 이름이 아닙니다.");
        }
        return nameToItem.get(keyName);
    }
```
😶‍🌫️ 제가 정확히 이해를 못했는데, 이렇게 메서드명을 수정하기를 제안해 주신걸까요?

Reply:
넵, 맞습니다. by는 굳이 안써도 될것 같아요 :)  검색 메서드가 그렇게 많지 않아서요